Status gui branch

lawwm opened the pull request:
Add GUI splitpane for note leaving status that @Btaykb is implementing.

**Warning**:
addressbook.json has been altered so it needs to be changed else it cannot be loaded. Find example implementation in 
`src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json`

**Justification:** 
Need to leave a note for person in status. Troublesome to write long note for each person especially if notes are similar. Therefore, allow storing status and giving same status for different people.

I think have some inkling of requirements but @Btaykb can change if want.

**Requirements:**
Add status to right pane.
Delete status from right pane. (when delete status, reset status of all persons who have that status?)
Add existing status in right pane to person stored in left pane.
Remove ability to add status to person directly.
Edit status (Optional)

Store status within each person but does not display in left pane.
Find on right pane using `findstatus` command.


![image](https://user-images.githubusercontent.com/41231584/159156070-c28fab80-7492-4a5f-b1e3-58f901cb0850.png)

### List all status: `liststatus`
Show a list of all statuses in ModuleMate Finder.


Format: `liststatus`

### Find a person's status: `findstatus`

Finds a person's status.

Format: `findstatus INDEX`

Format: `findstatus INDEX s/STATUS`
- Finds a status for the person at specified `INDEX`
- `INDEX` must be a **positive integer** 1, 2, 3, ...

Examples:
- `findstatus 1` finds the status of the 1st person in ModuleMate Finder.
- `findstatus 2` finds the status of the 2nd person in ModuleMate Finder.

Btaykb commented:
Nice work on the second pane! Just for clarification, for your feature, will the `comment` now be added to the newly added separate list (right pane) instead of the usual one (left pane with all the contacts)?

Because the way I implemented my `comment` feature is basically like `status` (i.e an additional field) that can be added/removed from a contact and then be displayed under their name. Based on my understanding of your feature, I'm thinking for mine to work with yours, maybe the `comment` will need to be added to the newly added right pane instead of to the `PersonCard` in the left pane (as it currently is).

lawwm commented:
> Nice work on the second pane! Just for clarification, for your feature, will the `comment` now be added to the newly added separate list (right pane) instead of the usual one (left pane with all the contacts)?
> 
> Because the way I implemented my `comment` feature is basically like `status` (i.e an additional field) that can be added/removed from a contact and then be displayed under their name. Based on my understanding of your feature, I'm thinking for mine to work with yours, maybe the `comment` will need to be added to the newly added right pane instead of to the `PersonCard` in the left pane (as it currently is).

I implemented comment as an attribute of status. I was imagining that you would have to add both "favourite" and "comment" to the panel on the right before you can add the combined status object to the left.

I don't think yours can work with mine just by adding to right pane since you kept comment as attribute of person. I think maybe just push yours first, it's easier to change from my side.

Maybe this makes sense as new requirements:
**Justification:** Annoying to type out comments repeatedly. Might repeat the same comment for different persons.

**Requirements:**

Save a comment on the right for future use.
`savecomment c/COMMENT`

Add comment from right pane to person on left pane.
`addcomment INDEX INDEX`

Add a comment to a person on the left.
`addcomment INDEX c/COMMENT`

Clear a comment from a person on the left.
`clearcomment INDEX`

Delete comment on the right.
List comment on right.
Find comment on right.
